pkg/kubectl/cmd/exec.go

```diff
@@ -161,6 +162,10 @@ func (p *ExecOptions) Complete(f *cmdutil.Factory, cmd *cobra.Command, argsIn []
 		}
 	}
 
+	if strings.HasPrefix(p.PodName, "pod/") {
```

Comment:
Something like this: https://github.com/openshift/origin/blob/master/pkg/cmd/cli/cmd/cancelbuild.go#L126-L147